stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/PopUtil.java

```diff
@@ -0,0 +1,213 @@
+        valueMap.put(EmailConstants.MESSAGE_CC, ccAddressArrayValue);
+        valueMap.put(EmailConstants.MESSAGE_BCC, bccAddressArrayValue);
+        valueMap.put(EmailConstants.MESSAGE_REPLY_TO, replyToAddressArrayValue);
+        valueMap.put(EmailConstants.MESSAGE_SUBJECT, subject);
+        valueMap.put(EmailConstants.MESSAGE_MESSAGE_BODY, messageBody);
+        valueMap.put(EmailConstants.MESSAGE_FROM, fromAddress);
+        valueMap.put(EmailConstants.MESSAGE_SENDER, senderAddress);
+        if (log.isDebugEnabled()) {
+            log.debug("[PopUtil][getMapValue] MapValue object created. Returning Object to PopClient");
+        }
+        return BallerinaValues.createRecordValue(EmailConstants.EMAIL_PACKAGE_ID, PopConstants.EMAIL, valueMap);
+    }
+
+    private static String getTextFromMimeMultipart(
+            MimeMultipart mimeMultipart)  throws MessagingException, IOException {
+        StringBuilder result = new StringBuilder();
+        int count = mimeMultipart.getCount();
+        for (int i = 0; i < count; i++) {
+            BodyPart bodyPart = mimeMultipart.getBodyPart(i);
+            if (bodyPart.isMimeType("text/plain")) {
+                result.append("\n").append(bodyPart.getContent());
+                break; // without break same text appears twice in my tests
+            } else if (bodyPart.isMimeType("text/html")) {
+                String html = (String) bodyPart.getContent();
+                // result = result + "\n" + org.jsoup.Jsoup.parse(html).text();
+                result.append("\n").append(html);
+            } else if (bodyPart.getContent() instanceof MimeMultipart) {
+                result.append(getTextFromMimeMultipart((MimeMultipart) bodyPart.getContent()));
+            }
+        }
+        return result.toString();
+    }
+
+    private static String getStringNullChecked(String string) {
+        return string == null ? "" : string;
+    }
+
+    /**
+     * Extracts port value and checks validity.
+     *
+     * @param longValue Port value extracted from the Config
+     * @return Port value as a string
+     */
+    private static String extractPortValue(long longValue, boolean isPop) {
+        if (log.isDebugEnabled()) {
+            log.debug("[PopUtil][extractPortValue] Extracting Port value");
+        }
+        if (longValue <= 0 || longValue > MAX_PORT) {
+            if (isPop) {
+                log.error("Invalid port number given in configuration. Setting default port "
+                        + PopConstants.DEFAULT_PORT_NUMBER);
+                return PopConstants.DEFAULT_PORT_NUMBER;
+            } else {
+                log.error("Invalid port number given in configuration. Setting default port "
+                        + ImapConstants.DEFAULT_PORT_NUMBER);
+                return PopConstants.DEFAULT_PORT_NUMBER;
+            }
+        } else {
+            return Long.toString(longValue);
+        }
```

Comment:
Shall we get rid of this else block and simply return without having it in an else block?

Reply:
Removed entire method for refactoring.